pkg/virt-controller/watch/vmi.go

```diff
@@ -590,13 +622,23 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 		}
 		c.recorder.Eventf(vmi, k8sv1.EventTypeNormal, SuccessfulCreatePodReason, "Created virtual machine pod %s", pod.Name)
 		return nil
+	} else {
+		if isTempPod(pod) && !isWaitForFirstConsumer {
```

Comment:
when we talk about a vmi being stuck indefinitely, this is specifically the place that occurs. We wait forever for the isWatiForFirstConsumer to become false without checking the status of that Pod (other than knowing it exists)